src/Exec_Top.cpp

```diff
@@ -19,105 +20,150 @@ Exec::RetType Exec_ParmInfo::Execute(CpptrajState& State, ArgList& argIn) {
+  else {
+    CpptrajFile* outfile = State.DFL().AddCpptrajFile(outname, desc);
+    if (outfile == 0) return CpptrajState::ERR;
+    mprintf("\tOutput to '%s'\n", outfile->Filename().full());
+    err = info.SetupTopInfo( outfile, parm, REF.RefPtr() );
+  }
+  return err;
+}
+
 // -----------------------------------------------------------------------------
 void Exec_BondInfo::Help() const {
-  mprintf("\t[%s] [<mask>]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print bond info for atoms in <mask> for specified topology (first by default).\n");
+  mprintf("\t[%s] [<mask1>] [<mask2>] [out <file>]\n", DataSetList::TopIdxArgs);
+  mprintf("  For specified topology (first by default) either print bond info for all\n"
+          "  atoms in <mask1>, or print info for bonds with first atom in <mask1> and\n"
+          "  second atom in <mask2>.\n");
 }
 
 Exec::RetType Exec_BondInfo::Execute(CpptrajState& State, ArgList& argIn) {
-  Topology* parm = State.DSL().GetTopByIndex( argIn );
-  if (parm == 0) return CpptrajState::ERR;
-  parm->PrintBondInfo( argIn.GetMaskNext() );
+  TopInfo info;
+  if (CommonSetup(info, State, argIn, "Bond info")) return CpptrajState::ERR;
+  std::string mask1 = argIn.GetMaskNext();
+  if (info.PrintBondInfo( mask1, argIn.GetMaskNext() )) return CpptrajState::ERR;
   return CpptrajState::OK;
 }
 // -----------------------------------------------------------------------------
 void Exec_AngleInfo::Help() const {
-  mprintf("\t[%s] [<mask>]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print angle info for atoms in <mask> for specified topology (first by default).\n");
+  mprintf("\t[%s] [<mask1>] [<mask2> <mask3>]\n\t[out <file>]\n", DataSetList::TopIdxArgs);
+  mprintf("  For specified topology (first by default) either print angle info for all\n"
+          "  atoms in <mask1>, or print info for angles with first atom in <mask1>,\n"
+          "  second atom in <mask2>, and third atom in <mask3>.\n");
 }
 
 Exec::RetType Exec_AngleInfo::Execute(CpptrajState& State, ArgList& argIn) {
-  Topology* parm = State.DSL().GetTopByIndex( argIn );
-  if (parm == 0) return CpptrajState::ERR;
-  parm->PrintAngleInfo( argIn.GetMaskNext() );
+  TopInfo info;
+  if (CommonSetup(info, State, argIn, "Angle info")) return CpptrajState::ERR;
+  std::string mask1 = argIn.GetMaskNext();
+  std::string mask2 = argIn.GetMaskNext();
+  if (info.PrintAngleInfo( mask1, mask2, argIn.GetMaskNext() )) return CpptrajState::ERR;
   return CpptrajState::OK;
 }
 // -----------------------------------------------------------------------------
 void Exec_DihedralInfo::Help() const {
-  mprintf("\t[%s] [<mask>] [and]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print dihedral info for atoms in <mask> for specified topology (first by default).\n"
-          "  If 'and' is specified dihedral must include all atoms specfied by <mask>.\n");
+  mprintf("\t[%s] [<mask1>] [<mask2> <mask3> <mask4>]\n\t[out <file>]\n", DataSetList::TopIdxArgs);
```

Comment:
Same comment as above -- Dihedrals can specify either 2, 3, or 4 masks.

Basically what this amounts to is having the defaults for every mask be `*`